### ISSUE TITLE
Search API

### ISSUE THREAD

**Issue by KidkArolis** [rvagg/ghutils#8]
Hi,

I was trying to use ghutils for github search, in particular, I need to use the `lister` to get all search results. However, the search results api returns a special object {totalItems, incomplete_results, items} and the lister doesn't handle it, because of the `data.length` check. I see several options:
### Option 1. Tweak the lister to handle search result objects (or objects in general), e.g. could check

``` js
util.isArray(data) ? retdata.push.apply(retdata, data) : retdata.push(data)
```

**UPDATE**: see #9 for the implementation

This way the lister would produce an array with [page1, page2, page3] search results, making it easy to create a search wrapper on top in my own code:

``` js
module.exports.search = async function search (auth, type, query, options) {
  var q = 'q=' + Object.keys(query).map(k => k + ':' + query[k]).join('+')
  var results = await list(auth, resolve('search/' + type + '?' + q), options)
  return results.reduce((list, page) => list.concat(page.items), [])
}
```
### Option 3. Create a separate `ghsearch` npm package

Basically, the functionality that `ghutils` already has (the lister and pagination handling) is exactly what I need in the search and it doesn't work only cause of https://github.com/rvagg/ghutils/blob/master/ghutils.js#L56, so would be nice to tweak that behaviour.
